### ISSUE TITLE
Idea (v. 202.5428.22) hangs after the plugin installation

### ISSUE THREAD

**Issue by dpishchukhin** [BashSupport-Pro/bashsupport-pro#26]
```
2020-06-12 10:38:01,478 [   2051]   WARN -        #com.intellij.idea.Main -  
java.lang.reflect.InvocationTargetException
	at java.desktop/java.awt.EventQueue.invokeAndWait(EventQueue.java:1373)
	at java.desktop/java.awt.EventQueue.invokeAndWait(EventQueue.java:1348)
	at com.intellij.idea.StartupUtil.runInEdtAndWait(StartupUtil.java:805)
	at com.intellij.idea.StartupUtil.access$000(StartupUtil.java:78)
	at com.intellij.idea.StartupUtil$1.invokeAndWait(StartupUtil.java:292)
	at com.intellij.util.ui.EdtInvocationManager.invokeAndWaitIfNeeded(EdtInvocationManager.java:68)
	at com.intellij.util.ui.UIUtil.invokeAndWaitIfNeeded(UIUtil.java:2269)
	at com.intellij.ide.b.a.Y.a(Y.java:260)
	at com.intellij.ide.b.l.h.b(h.java:313)
	at com.intellij.ide.b.l.h.a(h.java:131)
	at com.intellij.ide.b.o.a.a(a.java:99)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at com.intellij.idea.MainImpl.start(MainImpl.java:103)
	at com.intellij.idea.StartupUtil.startApp(StartupUtil.java:297)
	at com.intellij.idea.StartupUtil.prepareApp(StartupUtil.java:237)
	at com.intellij.ide.plugins.MainRunner.lambda$start$0(MainRunner.java:47)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.IllegalArgumentException: Argument for @NotNull parameter 'obj' of com/intellij/util/containers/SmartHashSet.remove must not be null
	at com.intellij.util.containers.SmartHashSet.$$$reportNull$$$0(SmartHashSet.java)
	at com.intellij.util.containers.SmartHashSet.remove(SmartHashSet.java)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at com.intellij.ide.b.a.O.d(O.java:168)
	at com.intellij.ide.b.a.O.<init>(O.java:78)
	at com.intellij.ide.b.a.c.<init>(c.java:251)
	at com.intellij.ide.b.a.a.a(a.java:52)
	at com.intellij.ide.b.a.a.b(a.java:45)
	at com.intellij.ide.b.l.h.c(h.java:318)
	at com.intellij.ide.b.a.Y.a(Y.java:267)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:303)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:776)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:727)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:746)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:417)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
```

**Comment by jansorg** [comment]
Thanks for the report!
Did you update the plugin or install it for the first time?
Is assume this happened after you clicked "Restart" -- is that correct?

**Comment by dpishchukhin** [comment]
I've tried both options. I've tried to delete the cache as well. Nothing helps.
I see this issue last 2 EAPs

**Comment by jansorg** [comment]
That's odd. It sounds like an issue with the EAP, but I'll try to reproduce this here.
Could you paste the information about your IDE and installed plugins, please? (Help->About->Ctrl+C)

**Comment by dpishchukhin** [comment]
```
IntelliJ IDEA 2020.2 EAP (Ultimate Edition)
Build #IU-202.5428.22, built on June 10, 2020
IntelliJ IDEA EAP User
Expiration date: July 10, 2020
Runtime version: 11.0.7+10-b931.3 x86_64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
macOS 10.15.5
GC: ParNew, ConcurrentMarkSweep
Memory: 1979M
Cores: 16
Non-Bundled Plugins: Lombook Plugin, com.thvardhan.gradianto, MavenRunHelper
```

**Comment by jansorg** [comment]
I was able to reproduce this, see https://youtrack.jetbrains.com/issue/IDEA-243468 for details. This is definitely something in the EAP, I can't fix this in BashSupport Pro. I'll leave this open until JetBrains has fixed the issue.

I was able to use BashSupport Pro with an existing license. But as soon as I removed it, it's all broken. I'm sorry that I can't fix this now.

**Comment by mabdelfattah** [comment]
I just bought the plugin yesterday and I've the same issue :(

**Comment by jansorg** [comment]
@mabdelfattah Thanks for buying a license!
You're using the 2020.2 eap, right? This shouldn't be happening with 2020.1 or earlier. 
As it's an IntelliJ eap things are unstable from time to time.

JetBrains seems to have fixed the issue and I expect the next IntelliJ eap to be out later this week. So that hopefully fixes this nasty issue.

**Comment by jansorg** [comment]
Today an IntelliJ eap was released, which is supposed to fix the hang at startup. 
BashSupport Pro on 2020.2 is usable now, but still has a few issues with find usages. I'll release an update for that soon.

**Comment by dpishchukhin** [comment]
I can confirm that the issue is not reproducible with the new EAP

**Comment by jansorg** [comment]
Thanks for confirming!
In the meanwhile I‘ve released 1.3.19.202 for the latest eap, which fixes all known compatibility issues with 2020.2.
